src/edu/csus/ecs/pc2/core/imports/LoadAccounts.java

```diff
@@ -397,4 +534,172 @@ public static void updateAccountsFromLoadAccountsFile(IInternalContest contest,
             contest.storeConfiguration(StaticLog.getLog());
         }
     }
+    
+    
+    
+    /**
+     * Create a list of updated accounts, if account in TSV file does not exist will create a new Account.
+     * 
+     * Intention is a list of all accounts in the TSV files that need to be updated or added.
+     * 
+     * File must have a header of field names.  The fields can appear in any order.
+     * 
+     * Reads TSV file, updates existing accounts in model, returns only a list
+     * of accounts that should be updated.
+     * 
+     * <P>
+     * 
+     * All accounts that are specified in the tsv file must exist, if an account
+     * does not exist then a IllegalTSVFormatException will be thrown.
+     * 
+     * <P>
+     * 
+     * 
+     * 1st line should contain the column headers.   Columns can appear in any order.
+     * Supported column headers are:
+     * <pre>
+     * account (required)
+     * alias
+     * displayname
+     * group (name)
+     * password
+     * permdisplay
+     * permlogin
+     * site (required)
+
+     * </pre>
+     * 
+     * @param filename
+     * @param existingAccounts 
+     * @param groupList
+     * @return an array of accounts
+     * @throws Exception
+     */
+    public Account[] fromTSVFileWithNewAccounts(String filename, Account[] existingAccounts, Group[] groupList) throws Exception  {
+        
+        /**
+         * Output accounts
+         */
+        Map<ClientId, Account> accountMap = new HashMap<ClientId, Account>();
+        
+        if (existingAccounts != null && existingAccounts.length > 0) {
+            for (int i = 0; i < existingAccounts.length; i++) {
+                existingAccountsMap.put(existingAccounts[i].getClientId(), existingAccounts[i]);
```

Comment:
If `existingAccountsMap` does not exist, the above line will throw an NPE.  Since this method is public, there's no way to guarantee that `existingAccountsMap` has been created before the method is called by some code (in the future if not in the present version).  The defensive thing to do would be to put a check/correction in front of the for-loop, like
```java
if (existingAccountsMap == null) {
  existingAccountsMap = new Map<ClientId,Account>():
}
```

Reply:
Never mind; I see that there is a initialization clause on `existingAccountsMap`, which is sufficient to ensure it is not null. Disregard comment.